test.py

```diff
@@ -1,5 +1,8 @@
 import os
 import torch
+import numpy as np
+import torch.nn.functional as F
+import matplotlib.pyplot as plt

```

Comment:
You can import matplotlib only if visualize is True.
So, you can move this line under if statement.

Reply:
But this change will cause `plt` import to be local, so other functions (whole call tree) will require `plt` as parameter to use. Do we actually need this?